src/test/java/org/candlepin/model/DeletedConsumerTest.java

```diff
@@ -16,41 +16,87 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import java.util.UUID;
+
 
 public class DeletedConsumerTest {
 
-    private DeletedConsumer dc = new DeletedConsumer("abcde", "10", "key", "displayname");
+    private DeletedConsumer deletedConsumer;
+
+    private String consumerUuid = UUID.randomUUID().toString();
+    private String consumerName = "consumer-name";
+    private String ownerId = "owner-id";
+    private String ownerKey = "owner-key";
+    private String ownerDisplayName = "owner-display-name";
+    private String principalName = "principal-name";
+
+    @BeforeEach
+    private void beforeEach() {
+        deletedConsumer = new DeletedConsumer(consumerUuid, ownerId, ownerKey, ownerDisplayName);
```

Comment:
Since we're updating this area, let's take the opportunity to detach from the injection-style constructor and use chained accessors instead so we can eventually remove those old constructors.

As a stylistic note: 
I'm not a huge fan of priming common test entities or objects in the initializers like this, as it makes the test suites fairly brittle and obscure as they get older and longer (see `ConsumerResourceTest` or `OwnerResourceTest` for an example of what this looks like extrapolated over a decade or more). It's fine for now, but will eventually become a maintenance headache.
What I like to see, instead, is a full test of the accessors and mutators from a base state. For example:

```
public void testGetSetConsumerUuid() {
  DeletedConsumer deletedConsumer = new DeletedConsumer();
  String testValue = "test_consumer_uuid";

  // Verify initial state is as we expect
  assertNull(deletedConsumer.getConsumerUuid());

  // Update the field, and ensure we're returning a self-ref
  DeletedConsumer out = deletedConsumer.setConsumerUuid(testValue);
  assertSame(out, deletedConsumer);

  // Verify final state is as expected/provided
  assertEquals(testValue, deletedConsumer.getConsumerUuid());
}
```

This can be broken up into separate tests if you'd prefer to be more strict about following unit test guidelines, but can get a little clunky as it's rather annoying to test the accessor without using the mutator or a purpose-built constructor (involves reflection and knowledge of the UUT's implementation, which is itself a no-no for good unit tests).

There's also an argument to be made that we should be testing mutators have no side effects on the other fields, but that requires much more infrastructure around both knowing and invoking the other accessors after every mutator. We've done this in the old DTO tests (check any of the 3.x branches for the API DTO tests for an example of this -- or maybe the manifest DTO tests in current master), but it's somewhat involved and clunky in its own right, and comes with its own baggage and headaches.

Reply:
This is great information! I like the example you gave of starting from a base state and then testing the accessors and mutators for that particular class member variable. I went ahead and implemented the rest of the tests in the same fashion and added a test for the constructors.

I can see how you could break up the accessors into their own tests without the mutators, but like you said, that would involve reflection to populate the private class fields and probably hard coding the field names and I think avoiding that is best.